js/src/scrollspy.js

```diff
@@ -110,6 +112,13 @@ class ScrollSpy extends BaseComponent {
     // TODO: on v6 target should be given explicitly & remove the {target: 'ss-target'} case
     config.target = getElement(config.target) || document.body
 
+    // TODO: v6 Only for backwards compatibility reasons. Use rootMargin only
+    config.rootMargin = config.offset ? `${config.offset}px 0px -30%` : config.rootMargin
+
+    if (typeof config.threshold === 'string') {
```

Comment:
Might be a dumb question but do we really need to handle string param?
In the unit tests if I do `'<div id="content" data-bs-threshold="[0,0.2,1]">',` it seems to work as well (haven't tried it in a real example).

Reply:
If it was on me and only, I wouldn't do it
 However, all our options are being parsed as strings too, so I am pretty sure that on the launch  day, we will face a new issue

Reply:
Oh I see. So let's keep things coherent, you're right :)